File: mqttclient/src/main/java/net/farsystem/mqttsngatek/mqttclient/MQTTConnack.java
package net.farsystem.mqttsngatek.mqttclient;

public class MQTTConnack {

    private int returnCode;
    private boolean isSessionPresent;

    public int getReturnCode() {
        return returnCode;
    }

    public void setReturnCode(int returnCode) {
        this.returnCode = returnCode;
    }

    public boolean isSessionPresent() {
        return isSessionPresent;
    }

    public void setSessionPresent(boolean sessionPresent) {
        isSessionPresent = sessionPresent;
    }

}

File: src/main/kotlin/net/farsystem/mqttsngatek/mqtt/MQTTConnack.kt
package net.farsystem.mqttsngatek.mqtt

data class MQTTConnack(
    val returnCode: MQTTReturnCode,
    val isSessionPresent: Boolean,
): MQTTMessage {
    override val type: MQTTMessageType = MQTTMessageType.CONNACK
}